ngx_pq_module.c

```diff
@@ -395,6 +397,7 @@ static ngx_int_t ngx_pq_res_tuples_ok(ngx_pq_save_t *s, ngx_pq_data_t *d, PGresu
             }
         }
     }
+    d->row_count += PQntuples(res);
```

Comment:
I think it would be better to place this line after the check for `d` being zero.